AlpsSDK/AlpsManager/AlpsManager.swift

```diff
@@ -30,19 +49,24 @@ open class AlpsManager: MatchMonitorDelegate, ContextManagerDelegate {
             "user-agent": "\(device.systemName) \(device.systemVersion)"
         ]
         AlpsAPI.customHeaders = headers
-        AlpsAPI.basePath = "https://api.matchmore.io/v5"
     }
     
     // MARK: - Match Monitor Delegate
     
-    func matchMonitor(monitor: MatchMonitor, didReceiveMatches: [Match]) {
-        
+    func didFind(matches: [Match], for device: Device) {
+        onMatch?(matches, device)
     }
     
     // MARK: - Context Manager Delegate
     
     func contextManager(manager monitor: ContextManager, didUpdateLocation: CLLocation) {
-        
+        mobileDevices.findAll { (result) in
+            guard case let .success(mobileDevices) = result else { return }
+            mobileDevices.forEach {
+                guard let deviceId = $0.id else { return }
+                self.locationUpdateManager.tryToSend(location: Location(location: didUpdateLocation), for: deviceId)
+            }
```

Comment:
Bit confusing for me. 

- you get all the mobile devices from repo.
- then you send each location of each mobile device.

It means you send location as a sort of batch ? 
Can't those mobile devices update their own location by themselves ?

after watching tryToSend function, this means that all mobiles devices will be at the self.lastLocation of this MAIN device ?

Reply:
Real life scenario is that you have only one device (your mobile phone).

But Rafał mentioned that we shouldn't disallow creating multiple mobile devices for now.